src/librustc_mir/monomorphize/partitioning.rs

```diff
@@ -585,7 +585,7 @@ fn merge_codegen_units<'tcx>(tcx: TyCtxt<'_, 'tcx, 'tcx>,
     // smallest into each other) we're sure to start off with a deterministic
     // order (sorted by name). This'll mean that if two cgus have the same size
     // the stable sort below will keep everything nice and deterministic.
-    codegen_units.sort_by_key(|cgu| cgu.name().clone());
+    codegen_units.sort_by_cached_key(|cgu| cgu.name().clone());
```

Comment:
Can't it be `sort_by(|a, b| a.name().cmp(b.name()))` to avoid heap allocations completely?